MaxText/tests/train_tests.py

```diff
@@ -0,0 +1,140 @@
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "steps=2",
+          "enable_checkpointing=False",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "synthetic": [  # tests base config with synthtic dataset
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "steps=2",
+          "enable_checkpointing=False",
+          "dataset_type=synthetic",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "pdb_lt_1": [  # tests base config with per_device_batch_size < 1
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "steps=2",
+          "enable_checkpointing=False",
+          "per_device_batch_size=0.25",
+          "ici_tensor_parallelism=4",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "int8": [  # tests base config with int8
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "quantization=int8",
+          "steps=2",
+          "enable_checkpointing=False",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "fp8": [  # tests base config with fp8
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "quantization=fp8",
+          "steps=2",
+          "enable_checkpointing=False",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "dropout": [  # tests base config with dropout
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          r"dataset_path=gs://maxtext-dataset",
+          "steps=2",
+          "enable_checkpointing=False",
+          "max_target_length=128",
+          "per_device_batch_size=1",
+          "dropout_rate=0.02",
+          r"tokenizer_path=../assets/tokenizer.llama2",
+      ],
+      "hf_input_pipeline": [  # test for train.py with TFDS c4, using HF input pipeline
+          None,
+          "configs/base.yml",
+          r"base_output_directory=gs://runner-maxtext-logs",
+          "run_name=runner_test",
+          "steps=2",
+          "enable_checkpointing=False",
+          "dataset_type=hf",
+          "hf_path=parquet",
+          r"hf_train_files=gs://maxtext-dataset/hf/c4/c4-train-00000-of-01637.parquet",
+          r"tokenizer_path=google-t5/t5-large",
+      ]
+  }
+
+  @pytest.mark.tpu_only
```

Comment:
What do you think of running separate @pytest for each config, e.g. something like 

AOT tests: https://github.com/AI-Hypercomputer/maxtext/blob/4699f4f2da2a3ff4956e89de8fb561550ecbbb16/MaxText/tests/train_compile_test.py#L27-L80

I prefer this since:
* Each test is more self contained, don't need to find the right config
* I think if a test fails in the current "for loop" implementation we won't get a clear error mesage as to which test really failed - we will get "test_tpu_config" failed and we will need to carefully scan the logs for the "Running TPU test for config...." message

Reply:
Done.